[Home] Dressing Room Updates

imgntn opened the pull request:
This PR does a few things:
- Adds the two photoreal avatars
- Adds collision hull with wider base
- Moves the small robot up to align with others
- Updates content on forum ATP so the next snapshot will have this stuff.

To test:
1. Reload Content (Clear all caches)
2. Visit forum
3. Hit the reset button
4. Go into the dressing room
5. Drop each avatar onto the dais and observe that it works

kevinmthomas-carpool commented:
testing

hifi-gustavo commented:
Build console output for commit: 0b867713cb416d11f4480e9f2e6376d0c7043674 available at: https://gist.github.com/c3f2195509dcd058af2ce263dfdf6b61<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7803.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7803.dmg

kevinmthomas-carpool commented:
## QA Test
**PR Build:** 7803
**Testing Architecture:** 64-bit PC Desktop
**Testing OS:** Windows 10
**Testing Locations:** forum
**Additional Test Results:** Successful test.  Reloaded content from ```localhost``` and then visited ```forum``` and hit the reset button and then and dropped each avatar unto the dais and confirmed it worked properly.  Screen shots attached.
**Successful Logout/Crash:** Successful logout.
![forum 1](https://cloud.githubusercontent.com/assets/7095767/14971515/f379f504-10a0-11e6-9a14-cb087718210c.png)
![avatar 1](https://cloud.githubusercontent.com/assets/7095767/14971519/f7c1f012-10a0-11e6-9341-e0b2234cd076.PNG)
![avatar 2](https://cloud.githubusercontent.com/assets/7095767/14971524/fb8c0cd2-10a0-11e6-907a-ca03a1e7fd99.PNG)
![avatar 3](https://cloud.githubusercontent.com/assets/7095767/14971525/fee796b2-10a0-11e6-878a-a53592f51809.PNG)
![avatar 4](https://cloud.githubusercontent.com/assets/7095767/14971527/01f7ea82-10a1-11e6-8142-b37e24d0352e.PNG)
![avatar 5](https://cloud.githubusercontent.com/assets/7095767/14971529/04f5af30-10a1-11e6-8e7d-9976b591632d.PNG)
![avatar 6](https://cloud.githubusercontent.com/assets/7095767/14971531/0838bd18-10a1-11e6-9437-dac9f7665504.PNG)

RyanDowne commented:
I replaced the Priscilla Av with a long sleeved shirt version. I think it might be worth one more tweak: can you try to make the spawning point of the dolls closer to the cabinet so they don't fall as far on rez. I think that is another reason they frequently fall over.

RyanDowne commented:
I think I'd also like to arrange them as - robot, being of light, stylized girl, will, priscilla, matthew

imgntn commented:
Changed the order and added a script to the dolls to stay upright after colliding with anything other than the dais.  Chatted with andrew re: the robot and there's apparently a bug with the rotational tensor being calculated off the visible geometry instead of the actual collision hull... so i made a custom 'box' shaped hull for the robot so it would stand like the others (if you use the 'box' type, it falls right down since the box is only as big as the visible geometry).  its a little wonky, we should decide whether to keep it.

hifi-gustavo commented:
Build console output for commit: d98732d5a1839724d2ae63825c664684c7f0d3f1 available at: https://gist.github.com/c3f2195509dcd058af2ce263dfdf6b61<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7803.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7803.dmg

kevinmthomas-carpool commented:
re-testing

kevinmthomas-carpool commented:
## QA Re-Test
**PR Build:** 7803
**Testing Architecture:** 64-bit PC Desktop
**Testing OS:** Windows 10
**Testing Locations:** forum
**Additional Test Results:** Successful test.  Order of avatars left to right, ```robot```, ```being of light```, ```stylized female```, ```will```, ```priscilla```, ```matthew```.  Upon arrival to ```forum```, ```being of light``` was loaded onto the dias as well as a miniature copy on the dresser with the other avatars.  ```robot``` upon picking up and release fell on its head and was able to manipulate as it is apparent has a larger collision hull which works better.  All of the other avatars when being picked up and release fell and landed on their feet except ```will``` who landed on his stomach.  Upon a tidy up event, re-tested ```will``` and he fell to his feet nicely.
**Successful Logout/Crash:** Successful logout.

ZappoMan commented on unpublishedScripts/DomainContent/Home/dressingRoom/transformer.js:
```diff
@@ -39,6 +39,20 @@
+        x: 1.8722,
+        y: 1.8197,
+        z: 0.3666
+    }
```
coding standard: missing trailing `;`

ZappoMan commented on unpublishedScripts/DomainContent/Home/dressingRoom/transformer.js:
```diff
@@ -65,18 +78,30 @@
-        // },
-
         findRotatorBlock: function() {
             print('transformer should find rotator block')
```
coding standard: missing trailing `;`

ZappoMan commented on unpublishedScripts/DomainContent/Home/dressingRoom/transformer.js:
```diff
@@ -125,14 +149,19 @@
                 print('TRANSFORMER IS A ROBOT')
                 dimensions = ROBOT_DIMENSIONS;
+            } else if (smallProps.modelURL.indexOf('priscilla') > -1) {
+                print('TRANSFORMER IS PRISCILLA')
```
coding standard: missing trailing `;`

ZappoMan commented on unpublishedScripts/DomainContent/Home/dressingRoom/transformer.js:
```diff
@@ -125,14 +149,19 @@
+                print('TRANSFORMER IS PRISCILLA')
+                dimensions = PRISCILLA_DIMENSIONS;
+            } else if (smallProps.modelURL.indexOf('matthew') > -1) {
+                print('TRANSFORMER IS MATTHEW')
```
coding standard: missing trailing `;`

ZappoMan commented on unpublishedScripts/DomainContent/Home/dressingRoom/wrapper.js:
```diff
@@ -9,48 +9,54 @@
-     }
-     var transformer = Entities.addEntity(transformerProps);
+    if (modelURL.indexOf('robot') > -1) {
+        print('THIS IS A ROBOT, GIVE IT A DIFFERENT SHAPE TYPE')
```
coding standard: missing trailing `;`

AndrewMeadows replied:
I was curious about the "trailing semicolon" coding standard since I did not remember it so I went looking for the specific item... and failed to find it:

https://github.com/highfidelity/hifi/wiki/Coding-Standard

Is the JS coding standard published elsewhere?

hifi-gustavo commented:
Build console output for commit: 6d3a3b0cc5c0039fa0ca1fbb569b557c0c7b89df available at: https://gist.github.com/c3f2195509dcd058af2ce263dfdf6b61<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7803.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7803.dmg
